example/loop_advection/loop_advection_package.cpp

```diff
@@ -0,0 +1,368 @@
+    IndexRange jb = pmb->cellbounds.GetFaceBoundsJ(IndexDomain::entire,face_dir_enum);
+    IndexRange kb = pmb->cellbounds.GetFaceBoundsK(IndexDomain::entire,face_dir_enum);
+
+    pmb->par_for(
+      "LoopAdvection::ProblemGenerator::init_B.x"+(face_dir)+"f", 0, vf.dim(5)-1, kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int k, const int j, const int i) {
+        const auto &coords = B.GetCoords(b);
+
+        //Compute the Curl of E
+        const int cart_dir_p1 = (cart_dir    )%3 + 1;
+        const int cart_dir_p2 = (cart_dir + 1)%3 + 1;
+
+        const Real dp1 = (cart_dir == 1) ? coords.dx1(i) :
+                        +(cart_dir == 2) ? coords.dx2(j) :
+                                           coords.dx3(k);
+        const Real dp2 = (cart_dir == 1) ? coords.dx1(i) :
+                        +(cart_dir == 2) ? coords.dx2(j) :
+                                           coords.dx3(k);
+
+        //First term: dEzdy, dExdz, or dEydx
+        const Real dEp2dp1 =(v( cart_dir_p2, b, A,  k, j, i)
+                            -v( cart_dir_p2, b, A,  k - (cart_dir_p1 == 3),
+                                                    j - (cart_dir_p1 == 2),
+                                                    i - (cart_dir_p1 == 1)))/dp1;
+
+        //Second term: dEydz, dEzdx, or dExdy
+        const Real dEp1dp2 =(v( cart_dir_p1, b, A, k, j, i)
+                            -v( cart_dir_p1, b, A, k - (cart_dir_p2 == 3),
+                                                   j - (cart_dir_p2 == 2),
+                                                   i - (cart_dir_p2 == 1)))/dp2;
+
+        v( cart_dir, b, B_face, b, k, j, i) = -(dEp2dp1 - dEp1dp2);
+      });
+  }
+
+}
+
+// Compute v x B to compute E, then take the curl to compute dBdt
+parthenon::TaskStatus calc_dBdt(parthenon::MeshBlock *pmb){
+
+  auto pkg = pmb->packages.Get("loop_advection_package");
+
+  const auto &v1 = pkg->Param<Real>("v1");
+  const auto &v2 = pkg->Param<Real>("v2");
+  const Real v3 = 0;
+
+  auto &v =  SparsePack<B_face,B_cell,E_cell,E_edge>::Make(pmd->meshblock_data.Get());
+
+  using dBdt_face = B_face;
+
+  IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::entire);
+  IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::entire);
+  IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::entire);
+
+  //First, compute the cell centered electric field 
+  {
+
+    pmb->par_for(
+      "LoopAdvection::ProblemGenerator:calc_E_cell", 0, vf.dim(5)-1, kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int k, const int j, const int i) {
+
+        //Average B from the faces to the cell centers
+        v(b, B_cell(0), k, j, i) = 0.5 * (v( 1, b, B_face, k, j, i) + v( 1, b, B_face,  k, j, i+1));
+        v(b, B_cell(1), k, j, i) = 0.5 * (v( 2, b, B_face, k, j, i) + v( 2, b, B_face,  k, j+1, i));
+        v(b, B_cell(2), k, j, i) = 0.5 * (v( 3, b, B_face, k, j, i) + v( 3, b, B_face,  k+1, j, i));
```

Comment:
When accessing a pack using a `variable_name` type, you need to instantiate the `variable_name` object by calling its constructor, so for instance this would need to change to 
```suggestion
        v(b, B_cell(0), k, j, i) = 0.5 * (v( 1, b, B_face(), k, j, i) + v( 1, b, B_face(),  k, j, i+1));
        v(b, B_cell(1), k, j, i) = 0.5 * (v( 2, b, B_face(), k, j, i) + v( 2, b, B_face(),  k, j+1, i));
        v(b, B_cell(2), k, j, i) = 0.5 * (v( 3, b, B_face(), k, j, i) + v( 3, b, B_face(),  k+1, j, i));
```
It looks like there are a number of other places a similar change is required.